### ISSUE TITLE
BUG: --autocorrect finds lint/if-nil-else in .cljc file reader conditionals even if it's not shown in normal mode

### ISSUE THREAD

**Issue by Cyrik** [NoahTheDuke/splint#48]
Me written this time, sorry for last time (also my assertion that :enabled got ignored was wrong. I didn't realize that lint/if-else-nil and lint/if-nil-else both exist and are different):

Splints `--autocorrect` finds and fixes issues in .cljc files that aren't issues in normal mode (and shouldn't be).

Repro src/example.cljc:
```clojure
(ns example.app)

(defn pick [d]
  (if (zero? d)
    #?(:clj :clj-branch :default :default-branch)
    d))
```

normal run:
```bash
bb -Sdeps '{:deps {io.github.noahtheduke/splint {:mvn/version "1.24.0"}}}' \
   -m noahtheduke.splint src/

Linting took 4ms, checked 1 files, 0 style warnings
```
--autocorrect run:

```
bb -Sdeps '{:deps {io.github.noahtheduke/splint {:mvn/version "1.24.0"}}}' \
   -m noahtheduke.splint --autocorrect src/

Replacing
(if (zero? d) nil d)
 with
(when-not (zero? d) d)
in src/example.cljc:4
```

**Comment by NoahTheDuke** [comment]
Thanks for the report!

**Comment by NoahTheDuke** [comment]
I took some time over my lunch to look at this, and I think I have a solution, but it's not ideal. Because of the way Splint is built, paired with how `--autocorrect` uses rewrite-clj, I don't know that I can avoid dataloss of some kind. I can treat a given `#?` reader conditional as the appropriate branch (`.clj` -> `:clj`, etc), but then if there is an autocorrection covering that code, the chosen branch will be used instead of the original reader-conditional.

I think the safest course of action is to skip `.cljc` files in autocorrect. That's not ideal either but the autocorrection system is already not great (overwrites comments/formatting), so here we can at least avoid data loss.

**Comment by Cyrik** [comment]
Agreed, it's tricky to get this correct.
I'm not sure you need to go as far as fully disabling it on cljc. It might be enough if you disable `if-nil-else` and `if-else-nil` autocorrections in cljc files.
I'm guessing some other rules would also interact badly with it, but these two would be the most common bad cases.
